li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -372,17 +458,44 @@ public Long safeOffset(TopicPartition tp) {
 
   @Override
   public void close() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    close(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
   }
 
   @Override
   public void close(long timeout, TimeUnit timeUnit) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    if (_consumers.isEmpty()) {
+      LOG.warn("No consumers to close for cluster group {}", _clusterGroup);
+      return;
+    }
+
+    LOG.info("Closing LiKafkaConsumer for cluster group {} in {} {}...", _clusterGroup, timeout, timeUnit);
```

Comment:
i dislike capitalization in log msgs as this makes grepping a problem. @jjkoshy ?

Reply:
Based on offline discussion, removed capitalization.